File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <cmath>
using namespace std;

int solution(int n) {
    int answer = 0;
    int arr[1000001]={0,};
    
    for(int i= 2;i<=n;i++){
        for(int j = 2;i*j<=n;j++){
            arr[i*j] = 1;
        }
    }
    for(int i = 2;i<=n;i++){
        if(arr[i] == 0){
            answer += 1;
        }
    }
    return answer;
}

File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def isPrimeNumber(x):
    if x < 2:
        return False
    for i in range(2,int((x**0.5))+1):
        if x % i == 0:
            return False
    return True
    
def solution(numbers):
    answer = 0
    arr = []
    for i in range(len(numbers)):
        arr.append(numbers[i])
    s4 = set()
    for i in range(1,len(numbers)+1):
        for j in permutations(arr, i):
            s = ""
            for st in list(j):
                s += st
            s4.add(int(s))
    s4 = list(s4)
    for i in range(len(s4)):
        if isPrimeNumber(s4[i]):
            answer+=1
    return answer